modules/clear_vacant_players.lua

```diff
@@ -101,13 +107,20 @@ function Public.remove_offline_players()
                                     end
                                 end
 
-                                local message = ({'main.cleaner', name})
+                                local message = ({ 'main.cleaner', name })
                                 local data = {
                                     position = pos
                                 }
                                 Alert.alert_all_players_location(data, message, nil, 20)
 
                                 e.die('neutral')
+                                if this.settings.dumped_inventories_do_not_expire == true then
+                                    -- Let's try to ensure that the corpse won't expire or get cleared.
+                                    -- That would defeat the purpose of this module.
+                                    local corpse = e.character_corpse
```

Comment:
I think what they are trying to access is something like this:
`for _, e in pairs(surface.find_entities_filtered({name = 'character-corpse'})) do`
`    if e and e.valid and e.character_corpse_player_index`
`        -- some func`
`    end`
`end`

Reply:
Yep something like that should work, though I would probably remember the position of spawned character (e) and search only around it? searching whole surface might be too expensive.

Also, this module's corpses don't have any player index asociated to them as it isn't any player's corpse.

Reply:
I _don't_ have it tested.
I think I'd rather exclude the `dumped_inventories_do_not_expire` feature for now, and perhaps have a go at it at a later date.

What's your preferred process?
- Do I add a commit reverting this and the Journey changes to this PR, and create a separate PR with the Journey changes?
- Or should I instead discard this PR and cherry-pick the parts into two new PRs?

---

I guess I thought it was LuaEntityPrototype typed 'character' in line 81, which would have [character_corpse](https://lua-api.factorio.com/latest/LuaEntityPrototype.html#LuaEntityPrototype.character_corpse).
Would it have a corpse on line 115 then, before being `die`d?
But not after, because on death it stops being a character, is that it?

Reply:
Prototype is the definition of entities, in this case it defines what happens when entity dies. That technically means that entity dies, and new entity is created on its spot, and this defines what entity it will be. So each entity has defined other entity that is just corpse, that comes for biters, buildings, trees leave stumps, and so on. Character corpse is a bit special in this case as it is actually interactable object rather than simple drawing on ground.

So no, even before .die() , it won't have corpse, it just has information, what corpse it will spawn when it gonna die.

It always goes that when entity dies, it just is removed from the game completely, and new entity for the corpse is spawned in. So there technically isn't any connection between the character and corpse at all (except when the character is owned by player, then the player's name and color gets transfered, but we work with characters that have no player), except the tick it is done at, and position.

Reply:
I see! Thank you for the explanation.

Reply:
On further exploration, this event actually should be able to solve this part, as it actually contains corpse entities generated by the killed entity. Though one would still have to do some clever filters to just distill the list to have corpses we just want (though not having player asociated with character might be probably enough).
https://lua-api.factorio.com/latest/events.html#on_post_entity_died

On the other hand, kinda not a fan of just another event that would be firing for _every killed entity_ and would have to be checked on all of them (even if we can exit the function early by checking if entity that died was character), we already have fair amount of performance hit from amount of biters and all that. And sadly the event filters work only once for whole scenario. But in theory this is what I would use for the functionality.

Reply:
Thanks very much for the feedback, Hanakocz.

I have shelved the corpse-decay improvement for now, I think it would be very useful to have for using in Journey
(especially as too often people are not playing near spawn and spawn is not defended)
but I don't have the focus for that work at the moment.